addOns/graphql/src/test/java/org/zaproxy/addon/graphql/GraphQlFingerprinterUnitTest.java

```diff
@@ -309,21 +309,28 @@ private static String errorResponse(String error, String field, boolean data) {
                 + " }";
     }
 
+    @SuppressWarnings("null")
     @ParameterizedTest
     @MethodSource("fingerprintData")
     void shouldFingerprintValidData(String graphqlImpl, String response) throws Exception {
         // Given
-        ExtensionAlert extensionAlert = mockExtensionAlert();
         nano.addHandler(new GraphQlResponseHandler(response));
         var fp = new GraphQlFingerprinter(UrlBuilder.build(endpointUrl));
+        DiscoveredGraphQlEngine[] discoveredEngine = new DiscoveredGraphQlEngine[1];
+        GraphQlFingerprinter.addEngineHandler(
+                (engine) -> {
+                    discoveredEngine[0] = engine;
+                });
         // When
         fp.fingerprint();
         // Then
-        ArgumentCaptor<Alert> alertArgCaptor = ArgumentCaptor.forClass(Alert.class);
-        verify(extensionAlert).alertFound(alertArgCaptor.capture(), any());
-        Alert alert = alertArgCaptor.getValue();
+        Alert.Builder builder = GraphQlFingerprinter.createFingerprintingAlert(discoveredEngine[0]);
+        Alert alert = builder.build();
```

Comment:
Is the builder needed? Could assign to the alert directly,
```suggestion
        Alert alert = GraphQlFingerprinter.createFingerprintingAlert(discoveredEngine[0]).build();
```